gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/GradleCacheExplorer.java

```diff
@@ -182,4 +186,9 @@ Optional<String> extractGroupAndArtifactFromUrl(Set<String> repoUrls, String url
     static GradleCacheExplorer getInstance() {
         return ApplicationManager.getApplication().getService(GradleCacheExplorer.class);
     }
+
+    // This should only ever be used by tests
```

Comment:
there is a typically a `@VisibleForTesting` annotation for this

Reply:
prefer to use a constructor for this i.e.

```
GradleCacheExplorer(Set<String> cachedUrls) {
  cache.set(cachedUrls);
}
```

Reply:
How will this work in the case where we are already using the constructor to load the cache - can I have multiple constructors? I essentially want a way to directly add to this classes cache as I dont think the runners have a real gradle cache I could use for tests

Reply:
you can have multiple constructors